### PR TITLE
Fix hvx_128 compilation plumbing issue

### PR DESCRIPTION
**Pull request by ronlieb**

### CODE REVIEW

File: src/HexagonOffload.cpp [hunk 1]
```diff
@@ -259,7 +259,16 @@ class InjectHexagonRpc : public IRMutator {
...
+
+        int result = 0;
+        string S64 = "${HEX_TOOLS}/bin/hexagon-clang hex.bc -fPIC -O3 -Wno-override-module -shared -o hex.so";
+        string S128 = S64 +" -mhvx-double";
+        Target target = hexagon_remote_target;
+        if (target.has_feature( Target::HVX_128))
+            result = system(S128.c_str());
+        else
+            result = system(S64.c_str());
+//          result = system("${HEX_TOOLS}/bin/hexagon-clang hex.bc -fPIC -O3 -Wno-override-module -shared -o hex.so");
```
Comment (pranavb-ca) [inline src/HexagonOffload.cpp]: May be you wanted to get rid of this comment?

**Comment by pranavb-ca** [comment]
LGTM modulo the removal of code that was commented out.

**Comment by ronlieb** [comment]
yes, i do want to remove it , thx

File: src/HexagonOffload.cpp [hunk 2]
```diff
@@ -284,7 +292,7 @@ class InjectHexagonRpc : public IRMutator {
 }
 
 Stmt inject_hexagon_rpc(Stmt s, const Target &host_target) {
-    Target target = hexagon_remote_target;
+    Target &target = hexagon_remote_target;
```
Comment (dsharletg) [inline src/HexagonOffload.cpp]: This seems wrong, it's going to modify the hexagon_remote_target, which is suppose to be a globally reused constant. 

Really, that should just not be a global, we should just change this code to be:

    Target target(Target::NoOS, Target::Hexagon, 32);
    for (Target::Feature i ...)

File: src/HexagonOffload.cpp [hunk 3]
```diff
@@ -259,7 +259,15 @@ class InjectHexagonRpc : public IRMutator {
 #else
         debug(1) << "Hexagon device code module: " << device_code << "\n";
         compile_module_to_llvm_bitcode(device_code, "hex.bc");
-        int result = system("${HEX_TOOLS}/bin/hexagon-clang hex.bc -fPIC -O3 -Wno-override-module -shared -o hex.so");
+
+        int result = 0;
+        string S64 = "${HEX_TOOLS}/bin/hexagon-clang hex.bc -fPIC -O3 -Wno-override-module -shared -o hex.so";
```
Comment (dsharletg) [inline src/HexagonOffload.cpp]: I'd prefer to write this like:

    string hex_command = "${HEX_TOOLS}... "
    if (target.has_feature(Target::HVX_128)) {
        hex_command += " -mhvx-double";
    }
    int result = system(hex_command.c_str());

File: src/HexagonOffload.cpp [hunk 4]
```diff
@@ -259,7 +259,15 @@ class InjectHexagonRpc : public IRMutator {
...
+        string S128 = S64 +" -mhvx-double";
+        Target target = hexagon_remote_target;
```
Comment (dsharletg) [inline src/HexagonOffload.cpp]: This should grab the target from the device_code module, and not copy it:

    if (device_code.target().has_feature(Target::HVX_128)) {
        ...
    }

File: src/HexagonOffload.cpp (hunk 2 above)
Comment (dsharletg) [inline src/HexagonOffload.cpp]: (And remove the global hexagon_remote_target)

**Comment by ronlieb** [comment]
we need to see the Target::HVX_64 or Target::HVX_128 feature when we are about invoke hexagon-clang, otherwise 128 byte mode does not compile correctly. 
Your suggestion will simply put it back the way it was.
so if you have a suggestion how to query for the attribute, let me know.

**Comment by dsharletg** [comment]
The target (with the HVX_64/HVX_128 flag set) gets passed through to the device_code module, so you can get the flag from there without having to go through the global target (which should be removed).

**Comment by ronlieb** [comment]
Pranav and i were looking some more, and the Module attribute for 128 byte mode is not being recognized by llvm

**Comment by ronlieb** [comment]
lets abandon this patch, and wait for a better fix.

**Comment by dsharletg** [comment]
OK, in the meantime, I'm going to fix the hexagon remote target so it's not a global.

**Pull request closed without merging**